src/test/java/info/batey/kafka/unit/MultipleBrokersIntegrationTest.java

```diff
@@ -0,0 +1,110 @@
+package info.batey.kafka.unit;
+
+import kafka.producer.KeyedMessage;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.TimeoutException;
+
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+
+public class MultipleBrokersIntegrationTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(MultipleBrokersIntegrationTest.class);
+    private KafkaUnit kafkaUnit;
+
+    @Before
+    public void before() {
+        kafkaUnit = new KafkaUnit(2);
+        kafkaUnit.startup();
+    }
+
+    @After
+    public void after() {
+        kafkaUnit.shutdown();
+    }
+
+    @Test
+    public void sendMessage() throws TimeoutException {
+        String topicName = "bla";
+        kafkaUnit.createTopic(topicName);
+        kafkaUnit.send(new KeyedMessage<String, String>(topicName, "hello"));
+        kafkaUnit.readKeyedMessages(topicName, 1);
+    }
+
+    @Test
+    public void movePartitions() throws TimeoutException, InterruptedException {
+        String topic = "bla";
+        int partitions = 50;
+        kafkaUnit.createTopic(topic, partitions);
+
+        createMsgsInTopic(topic, 1000);
+        kafkaUnit.readMessages(topic, 1000);
+
+        mapAllTopicPartitionsToSpecificBroker(topic, 2);
+
+        // Wait for reassignment to end
+        Thread.sleep(1000*10);
```

Comment:
Good call. I wrote another method which tries every second to see if all parititions replicas have been mapped to desired broker. It does for no more than 10 seconds. Have a look at next commit.